content/hacking-atom/sections/creating-a-grammar.md

```diff
@@ -3,203 +3,292 @@ title: Creating a Grammar
 
-* https://www.regular-expressions.info/tutorial.html provides a comprehensive regex tutorial
-* https://www.rexegg.com/regex-quickstart.html contains a cheat sheet for various regex expressions
-* https://regex101.com/ or https://regexr.com/ allows live prototyping
-* https://github.com/kkos/oniguruma/blob/master/doc/RE the docs for the Oniguruma regex engine
+#### The Parser
 
-Grammar files are written in the [CSON](https://github.com/bevry/cson#what-is-cson) or [JSON](https://www.json.org/) format. Whichever one you decide to use is up to you, but this tutorial will be written in CSON.
+Tree-sitter generates parsers based on [context-free grammars](https://en.wikipedia.org/wiki/Context-free_grammar) that are typically written in JavaScript. The generated parsers are C libraries that can be used in other applications as well as Atom.
 
-#### Create the Package
+They can also be developed and tested at the command line, separately from Atom. Tree-sitter has [its own documentation page](http://tree-sitter.github.io/tree-sitter/creating-parsers) on how to create these parsers. The [Tree-sitter GitHub organization](https://github.com/tree-sitter) also contains a lot of example parsers that you can learn from, each in its own repository.
 
-To get started, press <kbd class="platform-mac">Cmd+Shift+P</kbd><kbd class="platform-windows platform-linux">Ctrl+Shift+P</kbd> and start typing "Generate Package" to generate a new grammar package. Select "Package Generator: Generate Package," and you'll be asked for the path where your package will be created. Let's call ours `language-flight-manual`.
+Once you have created a parser, you need to publish it to [the NPM registry](https://npmjs.com) to use it in Atom. To do this, make sure you have a `name` and `version` in your parser's `package.json`:
 
-{{#tip}}
+```js
+{
+  "name": "tree-sitter-mylanguage",
+  "version": "0.0.1",
+  // ...
+}
+```
 
-**Tip:** Grammar packages should start with _language-_.
+then run the command `npm publish`.
 
-{{/tip}}
+#### The Package
 
-The default package template creates a lot of folders that aren't needed for grammar packages. Go ahead and delete the `keymaps`, `lib`, `menus`, and `styles` folders. Furthermore, in `package.json`, remove the `activationCommands` section. Now create a new folder called `grammars`, and inside that a file called `flight-manual.cson`. This is the main file that we will be working with - start by populating it with a [boilerplate template](https://gist.github.com/DamnedScholar/622926bcd222eb1ddc483d12103fd315). Now let's go over what each key means.
+Once you have a Tree-sitter parser that is available on npm, you can use it in your Atom package. Packages with grammars are usually named starting with _language_. You'll need a folder with a `package.json`, a `grammars` subdirectory, and a single `json` or `cson` file in the `grammars` directory, which can be named anything.
```

Comment:
Just to make it more clear, we should probably strengthen the wording:

> Packages with grammars are usually named starting with _language_.

to:

> Packages with grammars are, by convention, always named starting with _language_.

Even if things won't go wrong right now if they don't follow the convention, there are some things within Atom that started as convention but then behavior within Atom started to depend on that convention later, causing confusion.